homeassistant/components/tplink/__init__.py

```diff
@@ -102,22 +124,35 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         lights.extend(discovered_devices.lights)
         switches.extend(discovered_devices.switches)
 
-    forward_setup = hass.config_entries.async_forward_entry_setup
     if lights:
         _LOGGER.debug(
             "Got %s lights: %s", len(lights), ", ".join(d.host for d in lights)
         )
 
-        hass.async_create_task(forward_setup(entry, "light"))
-
     if switches:
         _LOGGER.debug(
             "Got %s switches: %s",
             len(switches),
             ", ".join(d.host for d in switches),
         )
 
-        hass.async_create_task(forward_setup(entry, "switch"))
+    # prepare DataUpdateCoordinators
+    hass.data[DOMAIN][COORDINATORS] = {}
+    for switch in switches:
+
+        try:
+            await hass.async_add_executor_job(switch.get_sysinfo)
+        except SmartDeviceException as ex:
+            _LOGGER.debug(ex)
+            raise PlatformNotReady from ex
```

Comment:
You should raise the Config entry exception.
```suggestion
            raise ConfigEntryNotReady from ex
```

Is it just possible to skip them for now assuming they will be discovered in the future again?

Reply:
honestly I simple copied the prior logic and do not want to also change this in this PR 🙈 
But yes, I agree with you, that this must be improved (_with an upcoming PR_)!